cmake/modules/zephyr_module.cmake

```
@@ -52,7 +52,9 @@ if(WEST)
   set(west_arg "--zephyr-base" ${ZEPHYR_BASE})
 endif()
 
-if(WEST OR ZEPHYR_MODULES)
+# Unittest may use modules too, we need to execute zephyr_module to
+# generate ZEPHYR_*_MODULE for existent modules.
+if(WEST OR ZEPHYR_MODULES OR UNITTEST)
```

Please revert.

Modules can either be provided through `west` (if installed) or the CMake variable `ZEPHYR_MODULES`.
Simply testing for `UNITTEST` doesn't ensure that code below is only executed when `west` is installed or `ZEPHYR_MODULES` is defined.

Instead you must ensure `WEST` and thus Zephyr modules are picked up through `west` (if not defined directly).
That is, the following line must be added here instead:
https://github.com/zephyrproject-rtos/zephyr/blob/9b851b2ab813b0bca588f0afeff7d5e878c0e789/cmake/modules/unittest.cmake#L4
```
include(west)
```

```suggestion
if(WEST OR ZEPHYR_MODULES)
```

---

Including `west` will just be used to define `WEST` variable and consequently having this block executed ...
Anyway, I have may missed something else. will change it since the change will become simpler.